tabbycat/draw/generator/common.py

```diff
@@ -40,16 +40,17 @@ class BaseDrawGenerator:
     requires_prev_results = False
     requires_rrseq = False
 
-    def __init__(self, teams, results=None, rrseq=None, **kwargs):
+    def __init__(self, teams, results=None, rrseq=None, teams_in_debate=2, **kwargs):
```

Comment:
[DRAFT COMMENT — this is partially developed thinking, but maybe it's useful or maybe you have thoughts on it already 🙃]

In fact… I don't think I'd recommend setting `teams_in_debate` as a constructor argument/instance attribute.

Given subclasses like `BaseBPDrawGenerator`, `BasePairDrawGenerator`, it seems to me that `teams_in_debate` should be an attribute of the class, not of the constructor.

If there's a need for a variable number of teams in a specific subclass, I'd create a mechanism for setting this in the constructor of that subclass, like:
``` py
class BaseManyTeamsDrawGenerator:
  def __init__(self, ..., teams_in_debate):
    self._teams_in_debate = teams_in_debate

  @property
  def teams_in_debate(self):
    return self._teams_in_debate
```
(or just set `self.teams_in_debate` in the constructor, understanding that it will override the class attribute.)

(If there's not really a need for subclasses anymore, then I guess ditch them?)